### PR TITLE
fix: pack: support network name overrides in bundle git tags

### PR DESCRIPTION
**Pull request by snissn**
## Related Issues
https://github.com/filecoin-project/lotus/issues/10185#issuecomment-1422864836

## Proposed Changes
add support for overrides 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by arajasek — COMMENTED** [review]
I think there's still one piece of information that needs to be passed through, which is the actors version being updated. As currently written, we'll respect the network overrides, but still set _all actor versions_ to the git tag, not just the one being packed.

As an example, `./pack.sh v9 dev/20230114-pre-rc.2 mainnet=v9.0.3 calibrationnet=v9.0.0 devnet=v9.0.3 devnet-wasm=v9.0.3 testing=v9.0.3 testing-fake-proofs=v9.0.3 butterflynet=v9.0.2` should only update v9 actors.

File: gen/bundle/bundle.go [hunk 1]
```diff
@@ -32,19 +34,46 @@ var EmbeddedBuiltinActorsMetadata []*BuiltinActorsMetadata = []*BuiltinActorsMet
...
-	// TODO: Re-enable this when we can set the tag for ONLY the appropriate version
-	// https://github.com/filecoin-project/lotus/issues/10185#issuecomment-1422864836
-	//if len(os.Args) > 1 {
-	//	for _, m := range metadata {
-	//		m.BundleGitTag = os.Args[1]
-	//	}
-	//}
+	var metadataSpecificVersion []*build.BuiltinActorsMetadata
+
+	// see ./build/actors/pack.sh
```
Comment (arajasek) [inline gen/bundle/bundle.go:50]: comment needs updating to include the actors version

File: gen/bundle/bundle.go [hunk 2]
```diff
@@ -32,19 +34,46 @@ var EmbeddedBuiltinActorsMetadata []*BuiltinActorsMetadata = []*BuiltinActorsMet
...
+		for _, m := range metadata {
+			if strings.HasPrefix(version, fmt.Sprintf("v%d", m.Version)) {
+				// correct version
+				override, ok := overrides[m.Network]
+				if ok {
+					m.BundleGitTag = override
+				} else {
+					m.BundleGitTag = os.Args[2]
+				}
+				fmt.Println("hi")
```
Comment (arajasek) [inline gen/bundle/bundle.go:70]: hi

**Review by arajasek — COMMENTED** [review]
Only issue I see with this is that it's gonna skip the metadata for those versions we _aren't_ setting, leading to them getting overwritten.

I think we can just drop the `metadataSpecificVersion`, and mutate the `metadata` object in place. That will modify it as needed, while leaving everything else untouched.

File: gen/bundle/bundle.go (hunk 2 above)
Comment (snissn) [reply gen/bundle/bundle.go:70]: hi 😆 🙊

**Review by arajasek — APPROVED** [review]

**Pull request merged**